### PR TITLE
IGNITE-23103 Sync RandomLruPageReplacementPolicy with Ignite 2

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-23103

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/ReplaceCandidate.java [hunk 1]
```diff
@@ -29,13 +29,13 @@
  */
 public class ReplaceCandidate {
     /** Partition generation saved in map, too old value means page may be safely cleared. */
-    private int gen;
+    private final int gen;
```
Comment (tkalkirill) [inline modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/ReplaceCandidate.java:32]: Why remove final?

Comment (sashapolo) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/ReplaceCandidate.java:32]: I didn't remove `final`, I added it

File: modules/page-memory/src/integrationTest/java/org/apache/ignite/internal/pagememory/tree/persistence/replacement/AbstractPageReplacementTest.java [hunk 2]
```diff
@@ -68,18 +68,17 @@
...
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 
 /** Integration tests for testing page replacement. */
-@ExtendWith({WorkDirectoryExtension.class, ConfigurationExtension.class})
-public class ItPageReplacementTest extends BaseIgniteAbstractTest {
+@ExtendWith(ConfigurationExtension.class)
+public abstract class AbstractPageReplacementTest extends IgniteAbstractTest {
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/page-memory/src/integrationTest/java/org/apache/ignite/internal/pagememory/tree/persistence/replacement/AbstractPageReplacementTest.java:81]: I'll move these tests in a separate PR, there are more of them, then just this

File: modules/page-memory/build.gradle [hunk 3]
```diff
@@ -59,6 +59,7 @@ dependencies {
     integrationTestImplementation project(':ignite-storage-api')
     integrationTestImplementation project(':ignite-failure-handler')
     integrationTestImplementation project(':ignite-file-io')
+    integrationTestImplementation project(':ignite-storage-page-memory')
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/page-memory/build.gradle:62]: It's a dependency for tests, I think we have a lot of similar places in other modules, which is considered fine

[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/page-memory/build.gradle:62]: I suggest not to introduce house into the project, why do you need this dependence?

File: modules/page-memory/src/integrationTest/java/org/apache/ignite/internal/pagememory/tree/persistence/replacement/AbstractPageReplacementTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/page-memory/src/integrationTest/java/org/apache/ignite/internal/pagememory/tree/persistence/replacement/AbstractPageReplacementTest.java:81]: For example: `Abstract class for testing page replacement using different strategies.`

File: modules/page-memory/build.gradle (hunk 3 above)
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/page-memory/build.gradle:62]: To import the `PersistentPageMemoryStorageEngine.ENGINE_NAME` constant. I still don't understand the problem, test dependencies are a separate graph. If there was a circular dependency, the project would not compile.

[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/page-memory/build.gradle:62]: Also, all the replacement policies are in the `page-memory` module

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**